[DeviceMesh][ROCm] skip ProcessGroup init test on ROCm because #ranks != #devices in CI

XilunWu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142386

**Summary**
Fixes #142361

Skip the DeviceMesh test since the test suite doesn't consider the case where `# ranks != # devices`.

**Test**
CI

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @naromero77amd

XilunWu committed:
[DeviceMesh][ROCm] skip ProcessGroup init test on ROCm because #ranks != #devices in CI

[ghstack-poisoned]

kwen2501 commented on test/distributed/test_device_mesh.py:
```diff
@@ -72,6 +75,7 @@ class DeviceMeshTest(DTensorTestBase):
     def world_size(self):
         return 4
 
+    @skip_if_rocm_multiprocess
```
nit: this is okay, but maybe consider `skip_if_lt_x_gpu(4)`?

XilunWu committed:
Update on "[DeviceMesh][ROCm] skip ProcessGroup init test on ROCm because #ranks != #devices in CI"


**Summary**
Fixes #142361

Skip the DeviceMesh test since the test suite doesn't consider the case where `# ranks != # devices`.

**Test**
CI

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o jeffdaily sunway513 jithunnair-amd pruthvistony ROCmSupport dllehr-amd jataylo hongxiayang naromero77amd

[ghstack-poisoned]

XilunWu commented:
update: using `skip_if_lt_x_gpu` is a better solution to address the issue.

XilunWu commented:
@pytorchbot merge
